caddyhttp/proxy/upstream.go

```diff
@@ -416,9 +423,21 @@ func (u *staticUpstream) healthCheck() {
 			}
 
 			if r, err := u.HealthCheck.Client.Do(req); err == nil {
-				io.Copy(ioutil.Discard, r.Body)
-				r.Body.Close()
-				unhealthy = r.StatusCode < 200 || r.StatusCode >= 400
+				if u.HealthCheck.ContentString != "" {
+					b, err := ioutil.ReadAll(r.Body)
```

Comment:
Done. I've also changed the logic a lot. Please take a look. I'll revert it you don't like it.